### PR TITLE
Task#15 - Feat: Product details components

### LINKED ISSUE #75: Task#15 - DevTask: Create Products details component
## OVERVIEW
- Using component in project buil UI for Products details components
- Update UI reponsive for product details

## DEMO
Reference photo:
image

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Closes #75 
## OVERVIEW

- Use existing components to build a new product details components
- Building ui through screen types ( responsive )
- Update Link component for heading and paragraph
- Mock data for description
- Tesing

## DEMO 

https://github.com/Agility-Internship/Duong-The-Hien-ReactJS/assets/113186169/c6f79b6c-7837-466e-a787-cb7c3bf47d99

### CODE REVIEW

File: practice-two/src/components/ProductDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+ * Product Details components.
+ *
+ *  A reusable component that displays product information in a details.
+ *
+ * @param product - The product product object.
+ * @param isFavoriteProduct - Boolean value indicating if the product is in favorites list.
+ * @returns {JSX.Element} Product Details content.
+ */
+const ProductDetails = (products = [], isFavoriteProduct = []) => {
+  const product = products.products;
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ProductDetails.jsx]: it's plural

File: practice-two/src/components/ProductDetails.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  return (
+    <section className="container min-h-screen flex justify-center items-center">
+      <div className="box bg-white rounded-2xl shadow-2xl p-10 m-2 w-96 grid grid-cols-1 lg:grid-cols-2 grid-rows-auto md:w-3/4 md:grid-rows-auto items-center">
+        {/* Images list */}
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ProductDetails.jsx]: remove redundant comments. update **ALL** cases

File: practice-two/src/components/ProductDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          <div className="border rounded-2xl">
+            <CardImage src={product.image} alt="This is a picture of the product" />
+          </div>
+          <div className="border rounded-2xl">
+            <CardImage src={product.image} alt="This is a picture of the product" />
+          </div>
+          <div className="border rounded-2xl">
+            <CardImage src={product.image} alt="This is a picture of the product" />
+          </div>
+        </div>
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ProductDetails.jsx:35]: image
 ??

File: practice-two/src/components/ProductDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+              {product.price}
+              &#8363;
+            </Typography>
+            {/* Displaying product description */}
+            <Typography className="leading-6" color="gray-300" size="md" variant="plain">
+              {product.description}
+            </Typography>
+            {/* Buy button */}
+            <Link
+              href="/products-card-details"
```
Comment (phattruong-agilityio) [inline practice-two/src/components/ProductDetails.jsx]: href ??

File: practice-two/src/components/ProductDetails.jsx (hunk 2 above)
Comment (TheHienDuong) [reply practice-two/src/components/ProductDetails.jsx]: Done

File: practice-two/src/components/ProductDetails.jsx (hunk 3 above)
Comment (TheHienDuong) [reply practice-two/src/components/ProductDetails.jsx:35]: Done

**Pull request merged**